homeassistant/components/firefly_iii/sensor.py

```diff
@@ -0,0 +1,138 @@
+"""Sensor platform for Firefly III integration."""
+
+from __future__ import annotations
+
+import logging
+
+from homeassistant.components.sensor import SensorEntity, SensorEntityDescription
+from homeassistant.components.sensor.const import SensorDeviceClass
+from homeassistant.components.todo import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddConfigEntryEntitiesCallback
+
+from .entity import FireflyBaseEntity
+
+_LOGGER = logging.getLogger(__name__)
+
+
+ACCOUNT_SENSORS: tuple[SensorEntityDescription, ...] = (
+    SensorEntityDescription(
+        key="account",
+        translation_key="account",
+        device_class=SensorDeviceClass.MONETARY,
+    ),
+)
+
+CATEGORY_SENSORS: tuple[SensorEntityDescription, ...] = (
+    SensorEntityDescription(
+        key="category",
+        translation_key="category",
+        device_class=SensorDeviceClass.MONETARY,
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: AddConfigEntryEntitiesCallback,
+) -> None:
+    """Set up the Firefly III sensor platform."""
+    coordinator = entry.runtime_data
+    entities: list[SensorEntity] = []
+
+    for account in coordinator.data.accounts:
+        entities.extend(
+            [
+                FireflyAccountEntity(
+                    coordinator=coordinator,
+                    entity_description=description,
+                    account=account,
+                )
+                for description in ACCOUNT_SENSORS
+            ]
+        )
+
+    for category in coordinator.data.category_details:
+        entities.extend(
+            [
+                FireflyCategoryEntity(
+                    coordinator=coordinator,
+                    entity_description=description,
+                    category=category,
+                )
+                for description in CATEGORY_SENSORS
+            ]
+        )
+
+    async_add_entities(entities)
+
+
+class FireflyAccountEntity(FireflyBaseEntity, SensorEntity):
+    """Entity for Firefly III account."""
+
+    def __init__(self, coordinator, entity_description, account) -> None:
+        """Initialize Firefly account entity."""
+        super().__init__(coordinator, entity_description)
+        self._account = account
+        self._attr_unique_id = f"{coordinator.config_entry.unique_id}_{entity_description.key}_{account.id}"
+        self._attr_name = account.attributes.name
```

Comment:
this is probably a dynamic account name from the users setup? So you can't use translations?